src/main/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperations.scala

```diff
@@ -5,13 +5,21 @@ import slick.ast.FieldSymbol
 import slick.jdbc.PostgresProfile.api._
 import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes.DataType.DataType
 import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes._
+import tech.cryptonomic.conseil.tezos.QueryProtocolTypes.Query
 import tech.cryptonomic.conseil.tezos.{TezosDatabaseOperations => TezosDb}
 
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
+/**
+  * Trait containing the interface for the QueryProtocol
+  */
+trait QueryProtocolOperations {
 
-object PlatformDiscoveryOperations {
+  def queryWithPredicates(tableName: String, query: Query)(implicit ec: ExecutionContext): Future[List[Map[String, Any]]]
+}
+
+object PlatformDiscoveryOperations extends QueryProtocolOperations {
```

Comment:
ok, renamed, also I think that it may become generic, depending on future implementations of next platforms, but then I would have to inject DBs and `TableQueries` and make it a class - but for now I would leave it as an object.